external-crates/move/crates/move-core-types/src/annotated_visitor.rs

```diff
@@ -68,135 +112,216 @@ pub trait Visitor {
 ///     Ok(())
 /// }
 /// ```
-pub trait Traversal {
+pub trait Traversal<'b, 'l> {
     type Error: From<Error>;
 
-    fn traverse_u8(&mut self, _value: u8) -> Result<(), Self::Error> {
+    fn traverse_u8(
+        &mut self,
+        _driver: &ValueDriver<'_, 'b, 'l>,
```

Comment:
Post review pondering: I think I get why they are immutable now, since in most cases you will just want to grab the bytes and do something with them as opposed to mutating internal state like in the vec/struct/enum drivers.

Reply:
Yeah, `ValueDriver` doesn't have any mutable state to modify in fact (whereas the other drivers track progress through their respective sub-structures). I considered making it mutable for consistency with other drivers, but stopped short because I didn't think the hassle of Rust's aliasing constraints on `&mut` were worthwhile.